src/mesh/difops.cxx

```diff
@@ -74,8 +74,9 @@ const Field3D Grad_par(const Field3D &var, DIFF_METHOD method, CELL_LOC outloc)
 *******************************************************************************/
 
 const Field3D Grad_parP(const Field3D &apar, const Field3D &f) {
+  ASSERT1(apar.getMesh() == f.getMesh());
+
   Mesh *mesh = apar.getMesh();
-  ASSERT1(mesh == f.getMesh());
```

Comment:
note that this is (non-significantly) slower - as getMesh is called 3 times, rather than twice ...

Reply:
I don't think it will really make any difference. If you've got optimisations turned on, the call to `getMesh` will be inlined and will be one or two instructions more at most. The worst case is 10 instructions without optimisations, including stack management from the non-inlined function call, none of which are more than a single clock cycle. If speed is really that crucial, then just turn checks off, and the whole line will vanish anyway.